### PR TITLE
Add join related options to stress test

### PR DESCRIPTION
**Pull request by vdimir**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Testing Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
* Add join related options to stress tests

### CODE REVIEW

File: docker/test/stress/stress [hunk 1]
```diff
@@ -34,6 +34,16 @@ def get_options(i):
     if i % 2 == 1:
         options.append(" --database=test_{}".format(i))
 
+    if i % 7 == 0:
+        options.append(" --client-option='join_use_nulls=1'")
+
+    if i % 2 * 7 == 0:
```
Comment (tavplubix) [inline docker/test/stress/stress:40]: AFAIK `i % 2 * 7` means `(i % 2) * 7`

**Review by tavplubix — APPROVED** [review]

File: docker/test/stress/stress (hunk 1 above)
Comment (vdimir) [reply docker/test/stress/stress]: Great catch!
Changed to `14` and `21`

**Comment by tavplubix** [comment]
Stress test (thread) - infinite recursion in `StorageMerge::getVirtuals`
Other stress tests passed. Let's run it once again to find new failures

**Review by alexey-milovidov — APPROVED** [review]

**Comment by tavplubix** [comment]
All stress tests passed

**Comment by tavplubix** [comment]
Functional stateless tests (ANTLR debug) - `Logical error: 'Can't find queue-0000000633 in the memory queue. It is a bug. Entry: format version: 4` - 

It's related to intersecting parts and merge cancellation.
Fwd from Telegram chat:
```
Alexander Sapin, [08.06.21 18:30]
https://clickhouse-test-reports.s3.yandex.net/25079/fc805a1ece07919ea5e230808070521e7b23d811/functional_stateless_tests_(antlr_debug).html#fail1
Can't find queue-0000000076 in the memory queue. It is a bug. Entry: format version: 4

Alexander Tokmakov, [08.06.21 18:32]
create_time: 2021-06-08 17:22:39
source replica: 9
block_id: 
merge
1_0_0_0
1_1_1_0
1_2_2_0
1_3_3_0
1_4_4_0
1_5_5_0
into
1_0_5_1
deduplicate: 0
part_type: Compact

Alexander Tokmakov, [08.06.21 18:37]
там ещё посередине 
2021.06.08 17:22:45.346105 [ 214 ] {} <Trace> test_8weahb.src_11 (58bc020f-9ccf-4dcf-98bc-020f9ccf0dcf): Created log entry /test/01154_move_partition_long_test_8weahb/s1/src_4226268275/log/log-0000000062 for merge 1_0_5_1

Alexander Tokmakov, [08.06.21 18:46]
А, ну да, мы выполнили мёрж потерянного куска, потому что другая реплика успела отменить этот мёрж у себя и назначить новый мёрж через дырку с таким же new_part_name, но другими source_parts

Alexander Tokmakov, [08.06.21 18:47]
и чекер удалил не ту entry, потому что было несколько entry с одинаковым new_part_name (старый невыполнимый мёрж и новый мёрж через дырку)

Alexander Tokmakov, [08.06.21 18:57]
а, даже не так, тут сложнее

Alexander Tokmakov, [08.06.21 18:58]
в этом случае в virtual parts вообще не было пересекающихся кусков

Alexander Tokmakov, [08.06.21 19:15]
быстрая реплика успела отменить невыполнимый мёрж, назначить новый с таким же new_part_name и выполнить его

медленная реплика в это время пыталась выполнить старый мёрж (про новый она ещё не знала т.к. ещё не спуллила лог, поэтому не видела пересекающихся кусков) и вместо мёржа сделала фетч куска, но это был уже кусок от нового мёржа

т.е. logical error в ActivePartsSet не спасло бы от этого рейса
```

**Pull request merged**